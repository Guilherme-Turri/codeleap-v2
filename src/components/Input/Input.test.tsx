/* eslint-disable testing-library/no-unnecessary-act */
import { act, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import { Input } from '.'
import { componentRender } from '../../render/componentRender'

describe('<Input />', () => {
  it('should render on screen', () => {
    const mockValues = {
      type: 'Email',
      name: 'Email',
      onChange: jest.fn(),
      error: '',
      value: 'email'
    }
    const { container } = componentRender(<Input {...mockValues} />)
    expect(screen.getByTestId('Input')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render on screen whith error message', () => {
    const mockValues = {
      type: 'text',
      name: 'Email',
      onChange: jest.fn(),
      error: 'Please fill in a valid email.',
      value: '',
    }
    componentRender(<Input {...mockValues} />)
    const input = screen.getByTestId('Input');
    act(() => {
      userEvent.type(input, 'user@');
      input.blur();

    });
    const errorMessage = screen.getByText('Please fill in a valid email.');
    expect(errorMessage).toBeInTheDocument();
  })
})