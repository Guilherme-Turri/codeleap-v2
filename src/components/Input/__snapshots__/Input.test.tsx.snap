// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render on screen 1`] = `
.c0 {
  outline: none;
  all: unset;
  border-bottom: 1px solid #00CBA9;
  width: 100%;
  padding: 0.625rem;
  box-sizing: border-box;
  color: #F3F4F5;
}

.c0::-webkit-input-placeholder {
  color: '#95A5A6'!important;
}

.c0::-moz-placeholder {
  color: '#95A5A6'!important;
}

.c0:-ms-input-placeholder {
  color: '#95A5A6'!important;
}

.c0::placeholder {
  color: '#95A5A6'!important;
}

<div>
  <input
    autocomplete="new-password"
    class="c0"
    data-testid="Input"
    placeholder="Email"
    type="Email"
    value="email"
  />
</div>
`;
